juju/juju
# Merge Juju SLA feature branch "rising-sun"

## Description of change

Merge `rising-sun` feature branch into develop.

> Why is this change needed?

This is a required feature for the upcoming Juju 2.2 release.

## QA steps

> How do we verify that the change works?

All `rising-sun` PRs have detailed QA steps; refer to these for details.

The `juju sla` command should report "unsupported" for all models where an SLA has not been set.

Contact @cmars for end-to-end testing of models with other SLA settings.

## Documentation changes

> Does it affect current user workflow? CLI? API?

Yes. A new command, `juju sla`, has been added. Contact @cmars for details on this command and other changes introduced by this feature branch.

## Bug reference

> Does this change fix a bug? Please add a link to it.

It's not a bug, it's a feature!
